### DISCUSSION TITLE
Difference of WebMVC's library and Webflux's library

### DISCUSSION THREAD

**Discussion by Jumple86** [Netflix/dgs-framework#1521] [Q&A]
The sentence from the [documentation](https://github.com/enisdenjo/graphql-ws/blob/master/PROTOCOL.md) of `graphql-ws` has put me in a difficulty in my project.
```
If there is already an active subscriber for an operation matching the provided ID, regardless of the operation type, the server must close the socket immediately with the event 4409: Subscriber for <unique-operation-id> already exists.
```
I think it means that should use 'unique-operation-id' to determine whether to send the 4409 event.

And I have examined the source code of `graphql-dgs-spring-webflux-autoconfigure.com.netflix.graphql.dgs.webflux.handlers.WebsocketGraphQLTransportWSProtocolHandler` and `graphql-dgs-subscriptions-websockets.com.netflix.graphql.dgs.subscriptions.websockets.WebsocketGraphQLTransportWSProtocolHandler` respectively. I found that the places where the decision to send the 4409 event is made have some slight differences.

In graphql-dgs-subscriptions-websockets, it seems that the decision to send the 4409 event([at line 128](https://github.com/Netflix/dgs-framework/blob/master/graphql-dgs-subscriptions-websockets/src/main/kotlin/com/netflix/graphql/dgs/subscriptions/websockets/WebsocketGraphQLTransportWSProtocolHandler.kt#L129)) is indeed made using the 'unique-operation-id' to determine whether to send the event.
However, in graphql-dgs-spring-webflux-autoconfigure, it seems that the same decision([at line 92](https://github.com/Netflix/dgs-framework/blob/master/graphql-dgs-spring-webflux-autoconfigure/src/main/kotlin/com/netflix/graphql/dgs/webflux/handlers/WebsocketGraphQLTransportWSProtocolHandler.kt#L93)) is made using the WebSocket session ID to determine whether to send the event.

I would like to know why there is such a difference between the two.

In my project, I am using 'graphql-dgs-webflux-starter', so naturally my 'WebsocketGraphQLTransportWSProtocolHandler' is using 'graphql-dgs-spring-webflux-autoconfigure'. And in my schema, there are two or more entry points for subscriptions.
If the frontend uses two or more entry points simultaneously, it will receive the 4409 event, even if only one WebSocket connection is established.

Is there any way to prevent receiving the 4409 event and use WebSocket normally to receive data?

**Comment by srinivasankavitha** [comment]
Thanks for the detailed description. This appears to be a bug in the webflux implementation. We do not use webflux stack internally and hence may have just avoided discovering this issue. 

We'll fix this in an upcoming release in the next month or so, but I don't think we can prioritize a fix immediately. Feel free to open a PR of this is blocking you currently.

**Reply by srinivasankavitha** [reply]
Created this issue: https://github.com/Netflix/dgs-framework/issues/1525
